internal/sauceignore/matcher.go

```diff
@@ -0,0 +1,61 @@
+package sauceignore
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
+)
+
+const commentPrefix = "#"
+const sauceignore = ".sauceignore"
+
+// ReadIgnoreFile reads .sauceignore file and creates ignore patters if .sauceignore file is exists.
+func ReadIgnoreFile(path string) ([]gitignore.Pattern, error) {
```

Comment:
I'd rather we hide `gitignore.Pattern` as a dependency than spread it across in our system 💡